### PR TITLE
Add GIT_CONFIG_* env vars to go_repository allow-list

### PR DESCRIPTION
**Pull request by mortenmj**
**What type of PR is this?**

Feature

**What package or component does this PR mostly affect?**

go_repository

**What does this PR do? Why is it needed?**

Adds the following environment variables to the go_repository allow-list:

- GIT_CONFIG
- GIT_CONFIG_GLOBAL
- GIT_CONFIG_SYSTEM
- GIT_CONFIG_NOSYSTEM

**Which issues(s) does this PR fix?**

Fixes https://github.com/bazelbuild/bazel-gazelle/issues/1790

**Other notes for review**

### CODE REVIEW

**Review by fmeum — APPROVED** [review]

File: internal/go_repository.bzl [hunk 1]
```diff
@@ -220,6 +220,10 @@ def _go_repository_impl(ctx):
         "http_proxy",
         "https_proxy",
         "no_proxy",
+        "GIT_CONFIG",
```
Comment (fmeum) [inline internal/go_repository.bzl:223]: Could you add a `# keep sorted` comment as the first entry of this list? That should have buildifier sort the chunks.

Comment (mortenmj) [reply internal/go_repository.bzl:223]: @fmeum do you mean the entire list, or the GIT_ part of it?

Comment (fmeum) [reply internal/go_repository.bzl:223]: The entire list. Buildifier should then sort the entries in each chunk delimited by comments. Makes it easier for us to recognize duplicates. Please also run buildifier on the file to check that it works.

Comment (mortenmj) [reply internal/go_repository.bzl:225]: Done

**Review by fmeum — APPROVED** [review]

**Pull request merged**